jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/LogGroupSelectorTable.kt

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+import com.intellij.ui.TableSpeedSearch
+import com.intellij.ui.TableUtil
+import com.intellij.ui.table.TableView
+import com.intellij.util.ui.ColumnInfo
+import com.intellij.util.ui.ListTableModel
+
+data class LogGroup(
+    var selected: Boolean = false,
+    var name: String? = null
+)
+
+class LogGroupSelectorTable : TableView<LogGroup>(model) {
+    init {
+        TableSpeedSearch(this)
+        TableUtil.setupCheckboxColumn(this, 0)
+        TableUtil.updateScroller(this)
+    }
+
+    fun populateLogGroups(selectedLogGroups: Set<String>, availableLogGroups: List<String>) {
+        val (selected, modelItems) = availableLogGroups.mapIndexed { index, logGroup ->
+            if (logGroup in selectedLogGroups) {
+                index to LogGroup(true, logGroup)
+            } else {
+                null to LogGroup(false, logGroup)
+            }
+        }.unzip()
+
+        listTableModel.items = modelItems
+        TableUtil.selectRows(this, selected.filterNotNull().toIntArray())
```

Comment:
nit: I thought there was a `ListTableModel`/`TableViewModel` thing to select based off of the model, but maybe not